packages/jazz-tools/src/coValues/deepLoading.ts

```diff
@@ -56,8 +53,12 @@ export function fulfillsDepth(depth: any, value: CoValue): FulfillsDepthResult {
       for (const [key, item] of Object.entries(value)) {
         if (map._raw.get(key) !== undefined) {
           if (!item) {
-            result = "unfulfilled";
-            continue;
+            if (hasReadAccess(map, key)) {
+              result = "unfulfilled";
+              continue;
+            } else {
+              return "unauthorized";
+            }
```

Comment:
Basically I've changed the logic to check for the permissions when the item is `null`

This way we can know if the item is null because it's loading or because it's not accessible.